pkg/cgroups/cgroups.go

```diff
@@ -123,12 +124,38 @@ func init() {
 	handlers[Memory] = getMemoryHandler()
 	handlers[Pids] = getPidsHandler()
 	handlers[Blkio] = getBlkioHandler()
+
+	// By Default no controller is available
+	availableControllersMap = make(map[string]bool)
+	availableControllersMap[CPU] = false
+	availableControllersMap[CPUset] = false
+	availableControllersMap[Memory] = false
+	availableControllersMap[Pids] = false
+	availableControllersMap[Blkio] = false
 }
 
 // getAvailableControllers get the available controllers
 func getAvailableControllers(exclude map[string]controllerHandler, cgroup2 bool) ([]controller, error) {
 	if cgroup2 {
-		return nil, fmt.Errorf("getAvailableControllers not implemented yet for cgroup v2")
+		controllers := []controller{}
+		subtreeControl := cgroupRoot + "/cgroup.subtree_control"
+		// rootless cgroupv2: check available controllers for current user , systemd or servicescope will inherit
+		if rootless.IsRootless() {
+			uid := rootless.GetRootlessUID()
+			subtreeControl = fmt.Sprintf("%s/user.slice/user-%d.slice/cgroup.subtree_control", cgroupRoot, uid)
```

Comment:
IIUC the controllers there are not always available for rootless users.
Reading the cgorup.subtree_control of Podman process itself might be better (it is still not robust, either, but at least better...)

Reply:
cc @giuseppe @kolyshkin

Reply:
We cant use subtree created by podman process as it wouldn't exist , if `podman info` is called prior to creating any container.

Reply:
also i think podman cleans up it entries as soon as running container exits. Not sure.

Reply:
yes the problem is that some controllers are not enabled for the rootless user (and we don't really know what controllers are enabled until we do the equivalent of `systemd-run --scope --user` with systemd).  So I agree with @AkihiroSuda that reading the current enabled controllers is a best effort way of doing it, which should be fine in most cases

Reply:
@giuseppe @AkihiroSuda Thank a lot . I have added changes, this is resolved in latest commit